layer/nrlf/core/json_duplicate_checker.py

```diff
@@ -0,0 +1,125 @@
+import json
+from collections import OrderedDict
+from typing import Dict, List, Tuple
+
+JsonPrimitive = str | int | float | bool | None
+type JsonValue = JsonPrimitive | JsonObject | JsonArray
+JsonPair = tuple[str, JsonValue]
+JsonObject = list[JsonPair]
+JsonArray = list[JsonValue]
+
+
+class DuplicateKeyChecker:
+    """JSON structure duplicate key detector.
+
+    Tracks duplicate keys by maintaining path context during traversal.
+    Paths are recorded in dot notation with array indices:
+    - Objects: parent.child
+    - Arrays: parent.array[0]
+    - Nested: parent.array[0].child[1].key
+    """
+
+    def __init__(self):
+        # Here a list of paths because the same key name could be at different levels
+        self.duplicate_keys_and_paths: OrderedDict[str, list[str]] = OrderedDict()
```

Comment:
It doesn't need to be an ordered dict. The problem is that when testing, the order is not ensured... So unit testing, but particularly integration tests would intermittently fail. Unit tests could be fixed by sorting the output which I do already, but that cannot be done for integration tests.